### PR TITLE
FIX: Modified thirdparty list: thirdparty type per default unknown

### PR DESCRIPTION
**Pull request by Tim-Otte**
# FIX: thirdparty type per default unknown

Modified the thirdparty list to filter the thirdparty type per default using the _unknown_ value.

### CODE REVIEW

**Comment by Tim-Otte** [comment]
PR to review

File: htdocs/societe/list.php [hunk 1]
```diff
@@ -322,7 +322,7 @@
 }
 
 if ($search_status=='') $search_status=1; // always display active thirdparty first
-
+if ($search_type_thirdparty=='') $search_type_thirdparty='0'; // use unknown if the thirdparty type is not set
```
Comment (eldy) [inline htdocs/societe/list.php]: I don't understand : 
if $search_type_thirdparty is '', then later nothing will be added as search criteria. So it does what we want to.

Comment (Tim-Otte) [reply htdocs/societe/list.php]: No, it doesn't. For an unknown reason, the default value is "Other", which is then also taken into account in the filter. The default value must be "Unknown" for the filter to work correctly.

Comment (Tim-Otte) [reply htdocs/societe/list.php]: I tested it a few times with the current develop branch and it only works correctly with this change. Feel free to test it yourself.

**Comment by Tim-Otte** [comment]
@eldy When will this PR be merged? I am already waiting more than 2 months...

**Comment by eldy** [comment]
I do not experience any trouble with develop version, neither in v10 or v11.
Can you describe how to repeoduce the bug ?

**Comment by Tim-Otte** [comment]
@eldy This is how you can reproduce this bug:

**Current behaviour (tested with v10.0.3):**

1. Go to Third parties > List
2. Show the "Third party type" column if not visible
3. Note: _Nothing is selected in the related select input_
4. Now enter something e.g. into the zip code input and hit enter
5. Only third parties with third party type "Other" are shown

**Expected behaviour:**
1. Go to Third parties > List
2. Show the "Third party type" column if not visible
3. Note: _Nothing is selected in the related select input_
4. Now enter something e.g. into the zip code input and hit enter
5. All third parties matching the filter the user added are shown, not only third parties with type "Other"

**TL;DR**:

The current default filter value for the third party type is "Other" and not "Unknown". Using the changes I added long time ago the default value for the third party type is "Unknown"...

**Comment by Tim-Otte** [comment]
@eldy I would be pleased if you would merge this pull request in a timely manner. This bug is very annoying if you actively use the table of business partners. I created this pull request more than three months ago. It's not that difficult to test two lines of code and then merge it.

**Comment by eldy** [comment]
Merging actors can validate a merge in a stable version only if there is a bug and we can validate that the fix removes really the bug without introducing more problem.
But for the moment, using the method to reproduce the bug that you describe, on any stable version (10.0.6, 11.0.1), there is no way to reproduce.
So in such situation (when a bug occurs only on some platforms and there is no way to reproduce it on already last released version), we let the ticket open to let other people investigates, add more information to help to understand what happen, or just to feedback or report if they also have the issue (for this reason having an issue open is better). Having more report helps often to find the "common denominator" that generates the bug.

Did you check with a last stable version (10.0.6 or 11.0.1) ?
Can you try to reproduce it on demo.dolibarr.org (demo / demo)  and provide screenshot if you succeed.
